generate-client.sh

```diff
@@ -30,9 +30,26 @@ then
 fi
 echo "[SUCCESS]"
 
+echo "Cloning source code repository from GitHub..."
+git clone --depth 1 --single-branch --branch staging $REPO_URL /repo
```

Comment:
@hgrsd How can I provide the `REPO_URL` like we do for `$INPUT_NAME` via `args` or `action.yml`? 🤔

Reply:
@djbeaumont Or maybe you can help me 😉

Reply:
So to do this you can add to the args array in `action.yaml`. However, surely there's a better way of doing this where the repo is already available to the action without needing to manually pull it. 🤔

Reply:
If the question is where does it come from - it does come from action.yaml, but the convention is that `name` in action.yaml becomes available inside the action as `INPUT_NAME`. (i.e. input to the action)

Reply:
Right, but how the name translates to the action yaml in target repo?
```
        with:
          path: https://rostering-integrations.staging.birdie.care/docs.json
          name: rostering-integrations
          repo_url: git@github.com:birdiecare/rostering-integrations.git
```
Like `repo_url` gonna be converted to `INPUT_REPO_URL` in shell script automatically?

Reply:
Yes I believe so 👍

Reply:
Is it possible to do this without requiring a new configuration parameter - looks like github makes the repo available as another env var already: https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables

Reply:
@MichalLytek I think I commented as you were marking the thread as resolved ☝️  looks like we can get the repo URL (or at least the path part) using an existing environment variable, so we don't have to require users of hatch to provide it.

Reply:
> However, surely there's a better way of doing this where the repo is already available to the action without needing to manually pull it.

I know but I don't have time for such research and the optimization can be done later 😉

Reply:
So we should be able to use `REPO_URL=git@github.com:${GITHUB_REPOSITORY}.git`?

Reply:
I wonder if the github action has permission to clone the repo 😄 
I think yes because we use the checkout action normally

Reply:
I hope so 😬 hopefully we do, as we have a token that has permission to push the packages back to the repo.